File: bindings/python/examples/how_tos/nft_collection/00_mint_issuer_nft.py
import os

from dotenv import load_dotenv
from iota_sdk import MintNftParams, Utils, Wallet, WalletOptions, utf8_to_hex, MetadataFeature

load_dotenv()

# In this example we will mint the issuer NFT for the NFT collection.

wallet = Wallet(WalletOptions(storage_path=os.environ.get('WALLET_DB_PATH')))

if 'STRONGHOLD_PASSWORD' not in os.environ:
    raise Exception(".env STRONGHOLD_PASSWORD is undefined, see .env.example")

wallet.set_stronghold_password(os.environ["STRONGHOLD_PASSWORD"])

# Sync wallet with the node
wallet.sync()

# Issue the minting transaction and wait for its inclusion
print('Sending NFT minting transaction...')
params = MintNftParams(
    immutable_metadata=MetadataFeature({'data': utf8_to_hex(
        "This NFT will be the issuer from the awesome NFT collection")}),
)


tx = wallet.mint_nfts([params])

wallet.wait_for_transaction_acceptance(
    tx.transaction_id)

print(
    f'Tx accepted: {os.environ["EXPLORER_URL"]}/transactions/{tx.transaction_id}')

transaction = tx.payload.transaction

for outputIndex, output in enumerate(transaction.outputs):
    # New minted NFT id is empty in the output
    if output["type"] == 6 and output["nftId"] == '0x0000000000000000000000000000000000000000000000000000000000000000':
        outputId = Utils.compute_output_id(
            tx.transaction_id, outputIndex)
        nftId = Utils.compute_nft_id(outputId)
        print(f'New minted NFT id: {nftId}')

File: bindings/nodejs/examples/how_tos/nft_collection/00_mint_issuer_nft.ts
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import {
    MintNftParams,
    NftId,
    NftOutput,
    utf8ToHex,
    Utils,
    Wallet,
} from '@iota/sdk';
require('dotenv').config({ path: '.env' });

// In this example we will mint the issuer NFT for the NFT collection.
//
// Make sure that `STRONGHOLD_SNAPSHOT_PATH` and `WALLET_DB_PATH` already exist by
// running the `how_tos/accounts_and_addresses/create-wallet` example!
//
// Rename `.env.example` to `.env` first, then run
// yarn run-example ./how_tos/nft_collection/00_mint_issuer_nft.ts
async function run() {
    try {
        for (const envVar of [
            'WALLET_DB_PATH',
            'STRONGHOLD_PASSWORD',
            'EXPLORER_URL',
        ]) {
            if (!(envVar in process.env)) {
                throw new Error(
                    `.env ${envVar} is undefined, see .env.example`,
                );
            }
        }

        // Create the wallet
        const wallet = await Wallet.create({
            storagePath: process.env.WALLET_DB_PATH,
        });

        // To sign a transaction we need to unlock stronghold.
        await wallet.setStrongholdPassword(
            process.env.STRONGHOLD_PASSWORD as string,
        );

        await wallet.sync();
        console.log(`Wallet synced!`);

        // Issue the minting transaction and wait for its inclusion
        console.log(`Sending NFT minting transaction...`);
        const params: MintNftParams = {
            immutableMetadata: utf8ToHex(
                'This NFT will be the issuer from the awesome NFT collection',
            ),
        };
        const transaction = await wallet.mintNfts([params]);

        // Wait for transaction to get included
        const blockId = await wallet.reissueTransactionUntilIncluded(
            transaction.transactionId,
        );
        console.log(
            `Block included: ${process.env.EXPLORER_URL}/block/${blockId}`,
        );

        transaction.payload.transaction.outputs.forEach(
            (output, outputIndex) => {
                if (output instanceof NftOutput) {
                    const nftOutput = output as NftOutput;

                    // New minted NFT id is empty in the output
                    if (
                        nftOutput.nftId ===
                        '0x0000000000000000000000000000000000000000000000000000000000000000'
                    ) {
                        const outputId = Utils.computeOutputId(
                            transaction.transactionId,
                            outputIndex,
                        );
                        const nftId: NftId = Utils.computeNftId(outputId);
                        console.log(`New minted NFT id: ${nftId}`);
                    }
                }
            },
        );
    } catch (error) {
        console.log('Error: ', error);
    }
    process.exit(0);
}

void run().then(() => process.exit());
